File: Sqrt(x)/Sqrt(x).cpp
class Solution {
public:
    int sqrt(int x) {
        if(!x || (x == 1)){
            return x;
        }
        
        double x0 = 1.0;
        double x1 = (double)x / x0 + x0;
        x1 /= 2;
        while(abs(x1 - x0) >= 1){
            x0 = x1;
            x1 = (double)x / x0 + x0;
            x1 /= 2;
        }
        return x1;
    }
};

File: Sqrt(x)/Sqrt(x).java
public class Solution {
    public int sqrt(int x) {
        double x0 = 0;
        double x1 = 1;
        
        while(Math.abs(x1 - x0) >= 1){
            x0 = x1;
            x1 = (x1 + x / x1) / 2;
        }
        
        return (int)Math.floor(x1);
    }
}
